animations/tween-animations.md

```diff
@@ -0,0 +1,726 @@
+only takes a single Animation object. To solve this problem,
+we’ll create our own Tween objects and explicitly calculate the
+values we need. Here’s the code:  (changes noted with highlights)
+
+<!-- skip -->
+{% prettify dart %}
+import 'package:flutter/animation.dart';
+import 'package:flutter/material.dart';
+
+class AnimatedLogo extends AnimatedWidget {
+  // We can make the Tweens static because we don't change them.
+  [[highlight]]static final _opacityTween = new Tween<double>(begin: 1.0, end: 0.1);[[/highlight]]
+  [[highlight]]static final _sizeTween = new Tween<double>(begin: 0.0, end: 300.0);[[/highlight]]
+
+  AnimatedLogo({Key key, Animation<double> animation})
+      : super(key: key, listenable: animation);
+
+  @override
+  Widget build(BuildContext context) {
+    final Animation<double> animation = listenable;
+    return new Center(
+      child: new Opacity([[/highlight]]
+        opacity: _opacityTween.evaluate(animation),[[/highlight]]
+        child: new Container(
+          margin: new EdgeInsets.symmetric(vertical: 10.0),
+          height: [[highlight]]_sizeTween.evaluate(animation)[[/highlight]],
+          width: [[highlight]]_sizeTween.evaluate(animation)[[/highlight]],
+          child: new FlutterLogo(),
+        ),
+      ),
+    );
+  }
+}
+
+class LogoApp extends StatefulWidget {
+  @override
+  _LogoAppState createState() => new _LogoAppState();
+}
+
+class _LogoAppState extends State<LogoApp> with TickerProviderStateMixin {
+  AnimationController controller;
+  Animation<double> animation;
+
+  initState() {
+    super.initState();
+    controller = new AnimationController(
+        duration: const Duration(milliseconds: 2000), vsync: this);
+    animation = [[highlight]]new CurvedAnimation(parent: controller, curve: Curves.easeIn);[[/highlight]]
+
+    animation.addStatusListener((status) {
+      if (status == AnimationStatus.completed) {
+        controller.reverse();
+      } else if (status == AnimationStatus.dismissed) {
+        controller.forward();
+      }
+    });
+
+    controller.forward();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new AnimatedLogo(animation: animation);
+  }
+
+  @override
+  dispose() {
+    super.dispose();
+    controller.dispose();
+  }
+}
+
+void main() {
+  runApp(new LogoApp());
+}
+{% endprettify %}
+
+You can see that the GrowTransition class was changed to encapsulate
+its own Tween objects, whose `.evaluate()` functions were called with
+the parent’s animation object to calculate the required size and
+opacity values.
+
+You can find the source for this example,
+[animate5,](https://raw.githubusercontent.com/InMatrix/animation_tutorial/master/lib/animate5.dart)
+on GitHub.
+
+## Next steps
+
+There are many more classes to explore, including specialized Tween
+classes, animations specific to Material Design, ReverseAnimation,
+physics simulations and `fling()` methods. This doc should have
```

Comment:
Maybe we can link to relevant API docs here. Some developers might be able to figure things out, and if not, hopefully they will ask about them and file issues.